### ISSUE TITLE
Windows nightly CPU build failed

### ISSUE THREAD

**Issue by peterjc123** [pytorch/pytorch#32817]
## 🐛 Bug

Error text:
```
[2376/2462] Linking CXX shared module caffe2\python\caffe2_pybind11_state.cp36-win_amd64.pyd
FAILED: caffe2/python/caffe2_pybind11_state.cp36-win_amd64.pyd 
cmd.exe /C "cd . && %BUILD_PREFIX%\Library\bin\cmake.exe -E vs_link_dll --intdir=caffe2\CMakeFiles\caffe2_pybind11_state.dir --rc=C:\PROGRA~2\WINDOW~4\10\bin\100162~1.0\x64\rc.exe --mt=C:\PROGRA~2\WINDOW~4\10\bin\100162~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MIB055~1\2017\COMMUN~1\VC\Tools\MSVC\1411~1.255\bin\Hostx64\x64\link.exe /nologo caffe2\CMakeFiles\caffe2_pybind11_state.dir\python\pybind_state.cc.obj caffe2\CMakeFiles\caffe2_pybind11_state.dir\python\pybind_state_dlpack.cc.obj caffe2\CMakeFiles\caffe2_pybind11_state.dir\python\pybind_state_nomni.cc.obj caffe2\CMakeFiles\caffe2_pybind11_state.dir\python\pybind_state_registry.cc.obj caffe2\CMakeFiles\caffe2_pybind11_state.dir\python\pybind_state_int8.cc.obj caffe2\CMakeFiles\caffe2_pybind11_state.dir\python\pybind_state_ideep.cc.obj  /out:caffe2\python\caffe2_pybind11_state.cp36-win_amd64.pyd /implib:lib\caffe2_pybind11_state.lib /pdb:caffe2\python\caffe2_pybind11_state.pdb /dll /version:0.0 /machine:x64 /ignore:4049 /ignore:4217 /INCREMENTAL:NO /machine:x64 /ignore:4049 /ignore:4217  %PREFIX%\libs\python36.lib lib\onnx_proto.lib lib\torch.lib lib\torch_cpu.lib lib\c10.lib ..\mkl\lib\mkl_intel_lp64.lib ..\mkl\lib\mkl_intel_thread.lib ..\mkl\lib\mkl_core.lib ..\mkl\lib\libiomp5md.lib lib\mkldnn.lib lib\libprotobuf.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib  && cd ."
LINK: command "C:\PROGRA~2\MIB055~1\2017\COMMUN~1\VC\Tools\MSVC\1411~1.255\bin\Hostx64\x64\link.exe /nologo caffe2\CMakeFiles\caffe2_pybind11_state.dir\python\pybind_state.cc.obj caffe2\CMakeFiles\caffe2_pybind11_state.dir\python\pybind_state_dlpack.cc.obj caffe2\CMakeFiles\caffe2_pybind11_state.dir\python\pybind_state_nomni.cc.obj caffe2\CMakeFiles\caffe2_pybind11_state.dir\python\pybind_state_registry.cc.obj caffe2\CMakeFiles\caffe2_pybind11_state.dir\python\pybind_state_int8.cc.obj caffe2\CMakeFiles\caffe2_pybind11_state.dir\python\pybind_state_ideep.cc.obj /out:caffe2\python\caffe2_pybind11_state.cp36-win_amd64.pyd /implib:lib\caffe2_pybind11_state.lib /pdb:caffe2\python\caffe2_pybind11_state.pdb /dll /version:0.0 /machine:x64 /ignore:4049 /ignore:4217 /INCREMENTAL:NO /machine:x64 /ignore:4049 /ignore:4217 %PREFIX%\libs\python36.lib lib\onnx_proto.lib lib\torch.lib lib\torch_cpu.lib lib\c10.lib ..\mkl\lib\mkl_intel_lp64.lib ..\mkl\lib\mkl_intel_thread.lib ..\mkl\lib\mkl_core.lib ..\mkl\lib\libiomp5md.lib lib\mkldnn.lib lib\libprotobuf.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:caffe2\python\caffe2_pybind11_state.cp36-win_amd64.pyd.manifest" failed (exit code 1120) with the following output:
   Creating library lib\caffe2_pybind11_state.lib and object lib\caffe2_pybind11_state.exp
pybind_state.cc.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) class std::unordered_map<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,struct std::hash<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > >,struct std::equal_to<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > >,class std::allocator<struct std::pair<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const ,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > > > > __cdecl caffe2::opt::getFakeFp16OpMapping(bool,bool)" (__imp_?getFakeFp16OpMapping@opt@caffe2@@YA?AV?$unordered_map@V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@V12@U?$hash@V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@2@U?$equal_to@V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@2@V?$allocator@U?$pair@$$CBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@V12@@std@@@2@@std@@_N0@Z) referenced in function "public: class std::unordered_map<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,struct std::hash<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > >,struct std::equal_to<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > >,class std::allocator<struct std::pair<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const ,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > > > > __cdecl pybind11::detail::argument_loader<bool,bool>::call<class std::unordered_map<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,struct std::hash<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > >,struct std::equal_to<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > >,class std::allocator<struct std::pair<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const ,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > > > >,struct pybind11::detail::void_type,class <lambda_99147820c862423944761d70f23a6720> &>(class <lambda_99147820c862423944761d70f23a6720> &)&& " (??$call@V?$unordered_map@V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@V12@U?$hash@V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@2@U?$equal_to@V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@2@V?$allocator@U?$pair@$$CBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@V12@@std@@@2@@std@@Uvoid_type@detail@pybind11@@AEAV<lambda_99147820c862423944761d70f23a6720>@@@?$argument_loader@_N_N@detail@pybind11@@QEHAA?AV?$unordered_map@V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@V12@U?$hash@V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@2@U?$equal_to@V?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@2@V?$allocator@U?$pair@$$CBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@V12@@std@@@2@@std@@AEAV<lambda_99147820c862423944761d70f23a6720>@@@Z)
caffe2\python\caffe2_pybind11_state.cp36-win_amd64.pyd : fatal error LNK1120: 1 unresolved externals

```

## To Reproduce

1. python setup.py build

## Expected behavior

Build succeed.

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0): master
 - OS (e.g., Linux): Windows
 - How you installed PyTorch (`conda`, `pip`, source): source
 - Build command you used (if compiling from source): python setup.py build
 - Python version: 3.5 / 3.6 / 3.7
 - CUDA/cuDNN version: NA
 - GPU models and configuration: NA
 - Any other relevant information:

## Additional context

The offending commit is something between 5e2311033ebc4bb6009f70db68e4d63ba73a1633 and c7df28a2a3de131ad4e1b284fd59a1d6febd61de.

**Comment by peterjc123** [comment]
Seems to be resolved by #32116, #32114 and #32112 according to https://ezyang.github.io/pytorch-ci-hud/build/pytorch-master. Could you please confirm, cc @ezyang?

**Comment by ezyang** [comment]
Not those PRs, but I did revert the breaking commit. We gucci.
